src/pit.js

```diff
@@ -132,11 +138,9 @@ PIT.prototype.get_counter_value = function(i, now)
 
     dbg_log("diff=" + diff + " dticks=" + diff_in_ticks + " value=" + value + " reload=" + this.counter_reload[i], LOG_PIT);
 
-    if(value < 0)
-    {
-        var reload = this.counter_reload[i];
-        value = value % reload + reload;
-    }
+    // could be too large after restore_state
+    var reload = this.counter_reload[i];
+    value = (value % reload + reload) % reload;
```

Comment:
We should emit a warning when the value is larger than the reload, and we can also get rid of the double-modulus.

```javascript
    var reload = this.counter_reload[i];

    if(value >= reload)
    {
        dbg_log("Warning: Counter" + i + " value " + value  + " is larger than reload " + reload, LOG_PIT);
        value %= reload;
    }
    else if(value < 0)
    {
        value = value % reload + reload;
    }
```